src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -398,6 +406,111 @@ class %s extends {$options['class']}
         return $code;
     }
 
+    private function isPreloadable(string $class): ?bool
+    {
+        if (array_key_exists($class, $this->preloadableCache)) {
+            return $this->preloadableCache[$class];
+        }
+
+        $this->preloadableCache[$class] = true; // prevent recursion
+        if (in_array($class, ['self', 'static', 'parent'], true)) {
+            return $this->preloadableCache[$class] = null;
+        }
+
+        try {
+            if (!class_exists($class) && !interface_exists($class) && !trait_exists($class)) {
+                return $this->preloadableCache[$class] = false;
+            }
+        } catch (\Error $e) {
+            return $this->preloadableCache[$class] = false;
+        }
+
+        $r = new \ReflectionClass($class);
+        if (!$r->isUserDefined()) {
+            return $this->preloadableCache[$class] = null;
+        }
+
+        // Prior to PHP 8.1, typehinted properties have to be auto-loadable.
+        if (\PHP_VERSION_ID >= 80100) {
+            return $this->preloadableCache[$class] = true;
+        }
+
+        // Before PHP 7.4, user can not define typehinted properties
+        if (\PHP_VERSION_ID >= 70400) {
+            foreach ($r->getProperties() as $p) {
+                if (!$this->isPreloadableType($p->getType())) {
+                    // do not return to warm the preloadableCache
+                    $this->preloadableCache[$class] = false;
+                }
+            }
+        }
+
+        // code bellow does not prevent the class to be preloaded, but is here to warm the preloadableCache
```

Comment:
typo: `below`, not `bellow`: https://prowritingaid.com/grammar/1000167/Bellow-vs-below%E2%80%94what-is-the-difference